app/controllers/DeclarationController.scala

```diff
@@ -70,10 +70,19 @@ class DeclarationController @Inject()(
     authorisedDataRequestAsync(ern, draftId) { implicit request =>
       withSubmitCreateMovementModel { submitCreateMovementModel =>
         service.submit(submitCreateMovementModel).flatMap {
-          response =>
+          case Right(response) =>
             logger.debug(s"[onSubmit] response received from downstream service ${response.downstreamService}: ${response.receipt}")
             val updatedAnswers = request.userAnswers.copy(hasBeenSubmitted = true, submittedDraftId = Some(response.submittedDraftId))
             saveAndRedirect(DeclarationPage, LocalDateTime.now(), updatedAnswers, NormalMode)
+          case Left(UnexpectedDownstreamDraftSubmissionResponseError(UNPROCESSABLE_ENTITY)) =>
+            // emcs-tfe returns an UNPROCESSABLE_ENTITY when downstream responds with RIM validation errors,
+            // so we redirect to DraftMovementController to display these errors
```

Comment:
Just a general question. Is it possible to get a 422 back and it _not_ be a RIM validation error?

Reply:
From emcs-tfe, as far as I know no. Play _shouldn't_ throw a 422 without us explicitly throwing it, and I don't think we're explicitly throwing it anywhere at the moment.

Reply:
Ah yeah, just checked the spec. I was thinking that EIS _could_ return a 422 despite not being a RIM validation. But that is not the case